src/pages/iou/request/MoneyRequestAttendeeSelector.tsx

```diff
@@ -0,0 +1,317 @@
+    attendees?: Attendee[] | typeof CONST.EMPTY_ARRAY;
+
+    /** The type of IOU report, i.e. split, request, send, track */
+    iouType: IOUType;
+
+    /** The action of the IOU, i.e. create, split, move */
+    action: IOUAction;
+};
+
+function MoneyRequestAttendeeSelector({attendees = CONST.EMPTY_ARRAY, onFinish, onAttendeesAdded, iouType, action}: MoneyRequestAttendeesSelectorProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const [searchTerm, debouncedSearchTerm, setSearchTerm] = useDebouncedState('');
+    const {isOffline} = useNetwork();
+    const personalDetails = usePersonalDetails();
+    const {didScreenTransitionEnd} = useScreenWrapperTranstionStatus();
+    const [betas] = useOnyx(ONYXKEYS.BETAS);
+    const [activePolicyID] = useOnyx(ONYXKEYS.NVP_ACTIVE_POLICY_ID);
+    const policy = usePolicy(activePolicyID);
+    const [isSearchingForReports] = useOnyx(ONYXKEYS.IS_SEARCHING_FOR_REPORTS, {initWithStoredValues: false});
+    const {options, areOptionsInitialized} = useOptionsList({
+        shouldInitialize: didScreenTransitionEnd,
+    });
+    const cleanSearchTerm = useMemo(() => debouncedSearchTerm.trim().toLowerCase(), [debouncedSearchTerm]);
+    const offlineMessage: string = isOffline ? `${translate('common.youAppearToBeOffline')} ${translate('search.resultsAreLimited')}` : '';
+
+    const isPaidGroupPolicy = useMemo(() => PolicyUtils.isPaidGroupPolicy(policy), [policy]);
+    const isCategorizeOrShareAction = [CONST.IOU.ACTION.CATEGORIZE, CONST.IOU.ACTION.SHARE].some((option) => option === action);
+
+    useEffect(() => {
+        Report.searchInServer(debouncedSearchTerm.trim());
+    }, [debouncedSearchTerm]);
+
+    const defaultOptions = useMemo(() => {
+        if (!areOptionsInitialized || !didScreenTransitionEnd) {
+            OptionsListUtils.getEmptyOptions();
+        }
+
+        const optionList = OptionsListUtils.getFilteredOptions(
+            options.reports,
+            options.personalDetails,
+            betas,
+            '',
+            attendees as Attendee[],
+            CONST.EXPENSIFY_EMAILS,
+
+            // If we are using this component in the "Submit expense" flow then we pass the includeOwnedWorkspaceChats argument so that the current user
+            // sees the option to submit an expense from their admin on their own Workspace Chat.
+            iouType === CONST.IOU.TYPE.SUBMIT && action !== CONST.IOU.ACTION.SUBMIT,
+
+            !isCategorizeOrShareAction,
+            false,
+            {},
+            [],
+            false,
+            {},
+            [],
+            !isCategorizeOrShareAction,
+            false,
+            false,
+            0,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            undefined,
+            iouType === CONST.IOU.TYPE.INVOICE,
+            action,
+            isPaidGroupPolicy,
+        );
```

Comment:
Let's move it to getAttendeesOptions as discussed in person

Reply:
Done in the latest commit!